core/services/ocr2/plugins/ocr2keeper/evm21/mercury/v02/request.go

```diff
@@ -0,0 +1,188 @@
+package v02
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/avast/retry-go/v4"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/ocr2/plugins/ocr2keeper/evm21/mercury"
+)
+
+const (
+	mercuryPathV02      = "/client?" // only used to access mercury v0.2 server
+	retryDelay          = 500 * time.Millisecond
+	totalAttempt        = 3
+	contentTypeHeader   = "Content-Type"
+	authorizationHeader = "Authorization"
+	timestampHeader     = "X-Authorization-Timestamp"
+	signatureHeader     = "X-Authorization-Signature-SHA256"
+)
+
+type MercuryV02Response struct {
+	ChainlinkBlob string `json:"chainlinkBlob"`
+}
+
+type client struct {
+	mercuryConfig mercury.MercuryConfigProvider
+	httpClient    mercury.HttpClient
+	lggr          logger.Logger
+}
+
+func NewClient(mercuryConfig mercury.MercuryConfigProvider, httpClient mercury.HttpClient, lggr logger.Logger) *client {
+	return &client{
+		mercuryConfig: mercuryConfig,
+		httpClient:    httpClient,
+		lggr:          lggr,
+	}
+}
+
+func (c *client) DoRequest(ctx context.Context, streamsLookup *mercury.StreamsLookup, pluginRetryKey string) (mercury.MercuryUpkeepState, mercury.MercuryUpkeepFailureReason, [][]byte, bool, time.Duration, error) {
+	resultLen := len(streamsLookup.Feeds)
+	ch := make(chan mercury.MercuryData, resultLen)
+	if len(streamsLookup.Feeds) == 0 {
+		return mercury.NoPipelineError, mercury.MercuryUpkeepFailureReasonInvalidRevertDataInput, [][]byte{}, false, 0 * time.Second, fmt.Errorf("invalid revert data input: feed param key %s, time param key %s, feeds %s", streamsLookup.FeedParamKey, streamsLookup.TimeParamKey, streamsLookup.Feeds)
+	}
+	for i := range streamsLookup.Feeds {
+		go c.singleFeedRequest(ctx, ch, i, streamsLookup)
```

Comment:
I think I've asked this before elsewhere in a different PR, but does it make sense to limit the number of concurrent single feed requests?